File: frontend/web/src/types/order.ts
import type { PageResult } from './product'

/** 订单状态映射（与后端 OrderStatus 对齐） */
export const ORDER_STATUS: Record<number, string> = {
  10: '待支付',
  20: '已支付',
  30: '待发货',
  40: '已发货',
  50: '已签收',
  60: '退款中',
  70: '已关闭',
  80: '已退款',
  90: '已完结',
  95: '回退失败-待人工',
}

/** 订单状态 tag 类型 */
export const ORDER_STATUS_TAG: Record<number, 'warning' | 'info' | 'primary' | 'success' | 'danger'> = {
  10: 'warning',
  20: 'info',
  30: 'primary',
  40: 'info',
  50: 'success',
  60: 'danger',
  70: 'info',
  80: 'danger',
  90: 'success',
  95: 'danger',
}

/** 订单（orders 表，分页记录） */
export interface Order {
  id: number
  orderNo: string
  userId: number
  storeId: number
  status: number
  warehouseStatus: number | null
  totalAmount: number
  couponAmount: number
  pointsDeduct: number | null
  pointsDeductAmount: number
  pointsEarned: number | null
  pointsSource: number | null
  payAmount: number
  commissionRate: number | null
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  remark: string | null
  trackingNo: string | null
  carrier: string | null
  shippedBy: number | null
  shippedAt: string | null
  paidAt: string | null
  finishedAt: string | null
  refundApprovedBy: number | null
  refundApprovedAt: string | null
  expireAt: string | null
  autoCloseStatus: number
  urgeCount: number
  urgedAt: string | null
  shipTimeoutWarned: number | null
  autoSignedAt: string | null
  version: number
  createdAt: string
  updatedAt: string
}

/** 订单明细（order_items 快照 + 规格 JSON 已解析） */
export interface OrderItem {
  skuId: number
  name: string
  specs: Record<string, string> | null
  image: string | null
  price: number
  qty: number
  subtotal: number
}

/** 订单详情 VO（订单头 + 明细 + 支付单） */
export interface OrderDetail {
  id: number
  orderNo: string
  userId: number
  storeId: number
  status: number
  statusDesc: string
  warehouseStatus: number | null
  totalAmount: number
  couponAmount: number
  pointsDeductAmount: number
  pointsEarned: number | null
  payAmount: number
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  remark: string | null
  trackingNo: string | null
  carrier: string | null
  paidAt: string | null
  shippedAt: string | null
  finishedAt: string | null
  expireAt: string | null
  createdAt: string
  payNo: string | null
  payStatus: number | null
  items: OrderItem[]
}

/** 物流轨迹（order_shipping_logs） */
export interface ShippingLog {
  id: number
  orderId: number
  status: string
  trackingNo: string | null
  carrier: string | null
  operatorId: number | null
  note: string | null
  createdAt: string
}

/** 订单分页查询参数 */
export type OrderPageQuery = {
  orderNo?: string
  userId?: number
  storeId?: number
  status?: number
  page: number
  size: number
}

export type { PageResult }

File: backend/src/main/java/com/herbaltea/module/order/entity/Order.java
package com.herbaltea.module.order.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import com.herbaltea.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * orders 表实体（订单，对齐 V1__schema.sql 权威结构）
 *
 * <p>状态机（{@link com.herbaltea.module.order.OrderStatus}）：
 * 10待支付 → 20已支付 → 30待发货 → 40已发货 → 50已签收 / 60退款中 / 70已关闭 /
 * 80已退款 / 90已完结 / 95回退失败-待人工。
 * 状态流转一律 {@code CAS(status, version)} 双条件更新（16.2 / 16.11）。
 *
 * <p>warehouse_status（总部发货状态）：1待接单 / 2待发货 / 3已发货 / 4已签收。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("orders")
public class Order extends BaseEntity {

    // ===== 状态机常量（与 OrderStatus 对齐） =====
    public static final int STATUS_PENDING_PAYMENT = 10;
    public static final int STATUS_PAID = 20;
    public static final int STATUS_PENDING_SHIPMENT = 30;
    public static final int STATUS_SHIPPED = 40;
    public static final int STATUS_SIGNED = 50;
    public static final int STATUS_REFUNDING = 60;
    public static final int STATUS_CLOSED = 70;
    public static final int STATUS_REFUNDED = 80;
    public static final int STATUS_COMPLETED = 90;

    /** 总部发货状态：待接单 */
    public static final int WH_READY = 1;
    /** 总部发货状态：待发货 */
    public static final int WH_TO_SHIP = 2;
    /** 总部发货状态：已发货 */
    public static final int WH_SHIPPED = 3;
    /** 总部发货状态：已签收 */
    public static final int WH_SIGNED = 4;

    /** 自动关单标记：未触发 */
    public static final int AUTO_CLOSE_NONE = 0;
    /** 自动关单标记：已自动关单 */
    public static final int AUTO_CLOSE_TIMEOUT = 1;
    /** 自动关单标记：用户已取消 */
    public static final int AUTO_CLOSE_USER = 2;

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 订单号（uk 唯一） */
    private String orderNo;

    /** 买家（users.id） */
    private Long userId;

    /** 业绩归属门店（stores.id，下单时选择） */
    private Long storeId;

    /** 状态机（10-95，见 OrderStatus） */
    private Integer status;

    /** 总部发货状态：1待接单/2待发货/3已发货/4已签收 */
    private Integer warehouseStatus;

    /** 商品总额 */
    private BigDecimal totalAmount;

    /** 优惠券抵扣 */
    private BigDecimal couponAmount;

    /** 券归属：0无券 / 1平台券（平台承担）/ 2本店券（店铺承担）（v28，V14 补列） */
    private Integer couponScope;

    /** 积分抵扣数量 */
    private Long pointsDeduct;

    /** 积分抵扣金额 */
    private BigDecimal pointsDeductAmount;

    /** 本单发放积分 */
    private Long pointsEarned;

    /** 积分来源：1门店营销 / 2平台活动 */
    private Integer pointsSource;

    /** 实付金额 */
    private BigDecimal payAmount;

    /** 平台佣金比例（下单时快照，如 0.0500 = 5%） */
    private BigDecimal commissionRate;

    /** 收货人（快照） */
    private String receiverName;

    /** 收货电话（快照） */
    private String receiverPhone;

    /** 收货地址（快照） */
    private String receiverAddress;

    /** 买家备注 */
    private String remark;

    /** 物流单号 */
    private String trackingNo;

    /** 快递公司 */
    private String carrier;

    /** 发货仓管（admin_users.id） */
    private Long shippedBy;

    /** 发货时间 */
    private LocalDateTime shippedAt;

    /** 支付时间 */
    private LocalDateTime paidAt;

    /** 完成时间 */
    private LocalDateTime finishedAt;

    /** 退款审批人 */
    private Long refundApprovedBy;

    /** 退款审批时间 */
    private LocalDateTime refundApprovedAt;

    /** 未支付自动关单时间（下单 + 30min，定时任务扫描） */
    private LocalDateTime expireAt;

    /** 0未触发 / 1已自动关单 / 2用户已取消 */
    private Integer autoCloseStatus;

    /** 催发货次数 */
    private Integer urgeCount;

    /** 最近催发货时间 */
    private LocalDateTime urgedAt;

    /** 总部 48h 超时预警已触发（防重复告警） */
    private Integer shipTimeoutWarned;

    /** 发货 15 天无签收自动完成时间 */
    private LocalDateTime autoSignedAt;

    /** 乐观锁（状态流转 CAS，16.11） */
    @Version
    private Integer version;
}
